### PR TITLE
lldp yang model

### PR DESCRIPTION
**Pull request by SouravSinhaRoy**
[https://github.com/open-traffic-generator/models-yang/blob/lldp/artifacts/open-traffic-generator-lldp.txt](https://github.com/open-traffic-generator/models-yang/blob/lldp/artifacts/open-traffic-generator-lldp.txt)

<pre>
module: open-traffic-generator-lldp
  +--rw lldps
     +--ro lldp-interface* [name]
        +--ro name     -> ../state/name
        +--ro state
           +--ro name?                     string
           +--ro counters
           |  +--ro frame-in?          otg-types:counter64
           |  +--ro frame-out?         otg-types:counter64
           |  +--ro frame-error-in?    otg-types:counter64
           |  +--ro frame-discard?     otg-types:counter64
           |  +--ro tlv-discard?       otg-types:counter64
           |  +--ro tlv-unknown?       otg-types:counter64
           |  +--ro frame-error-out?   otg-types:counter64
           +--ro lldp-neighbor-database
              +--ro lldp-neighbors
                 +--ro lldp-neighbor* [neighbor_id]
                    +--ro neighbor_id     -> ../state/neighbor_id
                    +--ro state
                    |  +--ro system-name?               string
                    |  +--ro system-description?        string
                    |  +--ro chassis-id?                string
                    |  +--ro chassis-id-type?           enumeration
                    |  +--ro neighbor_id?               string
                    |  +--ro age?                       uint32
                    |  +--ro last_update?               uint32
                    |  +--ro ttl?                       uint32
                    |  +--ro port-id?                   string
                    |  +--ro port-id-type?              enumeration
                    |  +--ro port-description?          string
                    |  +--ro management-address?        string
                    |  +--ro management-address-type?   string
                    +--ro custom-tlv* [custom-type oui oui-subtype]
                    |  +--ro custom-type    -> ../state/custom-type
                    |  +--ro oui            -> ../state/oui
                    |  +--ro oui-subtype    -> ../state/oui-subtype
                    |  +--ro state
                    |     +--ro custom-type?   uint32
                    |     +--ro oui?           string
                    |     +--ro oui-subtype?   string
                    +--ro capabilities* [name]
                       +--ro name     -> ../state/name
                       +--ro state
                          +--ro name?      enumeration
                          +--ro enabled?   boolean
</pre>

### CODE REVIEW

File: models/lldp/open-traffic-generator-lldp.yang [hunk 1]
```diff
@@ -0,0 +1,482 @@
...
+    are controlled by an open traffic generator (OTG) whose definition
+    is outside of the context of this module.";
+
+  revision 2022-08-09 {
+    description "Initial revision.";
+    reference "0.1.0";
+  }
+
+  // grouping statements
+  grouping lldp-top {
```
Comment (PrasenjitAdhikary) [inline models/lldp/open-traffic-generator-lldp.yang:29]: The new file showing LLDP yang tree is not reflected/added in the artifacts directory.

File: artifacts/open-traffic-generator-lldp.txt [hunk 2]
```diff
@@ -0,0 +1,45 @@
+module: open-traffic-generator-lldp
+  +--rw interface-members
```
Comment (PrasenjitAdhikary) [inline artifacts/open-traffic-generator-lldp.txt:2]: this name should start with lldp. Path could be lldp-interfaces/lldp-interface[name=lldp_name]

File: artifacts/open-traffic-generator-lldp.txt [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        |  +--ro counters
+        |     +--ro frame-in?          otg-types:counter64
+        |     +--ro frame-out?         otg-types:counter64
+        |     +--ro frame-error-in?    otg-types:counter64
+        |     +--ro frame-discard?     otg-types:counter64
+        |     +--ro tlv-discard?       otg-types:counter64
+        |     +--ro tlv-unknown?       otg-types:counter64
+        |     +--ro frame-error-out?   otg-types:counter64
+        +--ro lldp-neighbors
+           +--ro lldp-neighbor* [neighbor_id]
```
Comment (PrasenjitAdhikary) [inline artifacts/open-traffic-generator-lldp.txt:16]: How to retrieve these info using OTG ? What are the corresponding api/options in OTG model - could not find any match.

File: models/lldp/open-traffic-generator-lldp.yang [hunk 4]
```diff
@@ -0,0 +1,482 @@
...
+          config false;
+
+          description
+            "Operational state of an individual LLDP interface.";
+
+          uses lldp-state;
+
+          container counters {
+            description
+              "Counters of an individual LACP peer.";
```
Comment (PrasenjitAdhikary) [inline models/lldp/open-traffic-generator-lldp.yang:81]: LLDP instead of LACP

File: models/lldp/open-traffic-generator-lldp.yang [hunk 5]
```diff
@@ -0,0 +1,482 @@
...
+        description
+          "Each LLDP interface is identified by an arbitrary string
+          identifier.";
+
+        leaf name {
+          type leafref {
+            path "../state/name";
+          }
+          description
+            "Reference to the LLDP interfce  member's name, acting as a key of the
```
Comment (PrasenjitAdhikary) [inline models/lldp/open-traffic-generator-lldp.yang:67]: LLDP interface or neighbor? How it is reflected in the OTG model?

File: artifacts/open-traffic-generator-lldp.txt [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+              +--ro custom-tlv* []
+              |  +--ro custom-type?   -> ../state/custom-type
+              |  +--ro oui?           -> ../state/oui
+              |  +--ro oui-subtype?   -> ../state/oui-subtype
+              |  +--ro state
+              |     +--ro custom-type?   uint32
+              |     +--ro oui?           string
+              |     +--ro oui-subtype?   string
+              |     +--ro tlv-value?     string
+              +--ro capabilities* []
```
Comment (PrasenjitAdhikary) [inline artifacts/open-traffic-generator-lldp.txt:41]: should be a key list i.e. capabilities*[name]

File: artifacts/open-traffic-generator-lldp.txt [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+              |  +--ro neighbor_id?               string
+              |  +--ro age?                       uint32
+              |  +--ro last_update?               uint32
+              |  +--ro ttl?                       uint32
+              |  +--ro port-id?                   string
+              |  +--ro port-id-type?              enumeration
+              |  +--ro port-description?          string
+              |  +--ro management-address?        string
+              |  +--ro management-address-type?   string
+              +--ro custom-tlv* []
```
Comment (PrasenjitAdhikary) [inline artifacts/open-traffic-generator-lldp.txt:32]: looks custom-type, oui and oui-subtype are key for the custom-tlv

File: models/lldp/open-traffic-generator-lldp.yang (hunk 1 above)
Comment (SouravSinhaRoy) [reply models/lldp/open-traffic-generator-lldp.yang:29]: change submitted

File: artifacts/open-traffic-generator-lldp.txt (hunk 2 above)
Comment (SouravSinhaRoy) [reply artifacts/open-traffic-generator-lldp.txt]: names updated

File: models/lldp/open-traffic-generator-lldp.yang (hunk 4 above)
Comment (SouravSinhaRoy) [reply models/lldp/open-traffic-generator-lldp.yang]: changed

File: artifacts/open-traffic-generator-lldp.txt (hunk 6 above)
Comment (SouravSinhaRoy) [reply artifacts/open-traffic-generator-lldp.txt]: added key list

File: artifacts/open-traffic-generator-lldp.txt (hunk 7 above)
Comment (SouravSinhaRoy) [reply artifacts/open-traffic-generator-lldp.txt]: added key list

File: models/lldp/open-traffic-generator-lldp.yang (hunk 5 above)
Comment (SouravSinhaRoy) [reply models/lldp/open-traffic-generator-lldp.yang:67]: lldp  configured interface name

File: artifacts/open-traffic-generator-lldp.txt (hunk 3 above)
Comment (SouravSinhaRoy) [reply artifacts/open-traffic-generator-lldp.txt:16]: please check the latest,in redocly viewer we can see the states changes

**Review by PrasenjitAdhikary — APPROVED** [review]
LGTM

**Review by apratimmukherjee — COMMENTED** [review]
Have a couple of comments . ( Including one typo )

File: models/lldp/open-traffic-generator-lldp.yang [hunk 8]
```diff
@@ -0,0 +1,484 @@
...
+
+           leaf oui-subtype {
+             type string;
+             description
+              "The organizationally defined subtype field shall contain a unique subtype value assigned by the 
+               defining organization.";
+           }
+
+           leaf tlv-value {
+             type string;
```
[the remark this answers is not in the document]
Comment (SouravSinhaRoy) [reply models/lldp/open-traffic-generator-lldp.yang]: oc doesn't specify ylv-value so removed it

**Pull request merged**